### PR TITLE
✨ [DRAFT] (Optionally?) Autoconvert Archive Mods to REDmods

### PR DESCRIPTION
**Pull request by effs**
- [ ] Any regular Archive mod that would be installed normally moved to canonical REDmod location
  - [ ] Also install stuff that may not be valid, show warning as usual (may need new copy?)
- [ ] Any extra Archive use converted also
- [x] info.json generated
  - [x] reasonable(?) 'mod name' for the mods that don't have one, incl. path (maybe something better than current fake name)
  - [x] I guess just better parsing of mod info
  - [x] version generated with build tag `+V2077RED` (no precedence implication) ~~something like `v2077-redmod` pre-release tag so it's still SemVer~~
- [x] Notification shown for every autoconverted mod

- [ ] mod type set to REDmodArchive (or whatever)
- [ ] mod attribute set to indicate autoconvert?

- [ ] Setting or something to let user control whether to enable this or not?
  - [ ] This requires reinstalling things, doesn't it? I guess that'll be manual for now - warn about it

- [x] Trivial feature flag implementation in features.ts, the flag set is passed to installers
- [x] Support overriding in individual examples

🏷️ REDmodname = dir, only 1 version installable

Because the name and dir are linked, it's not possible to have two versions of the same (by name) REDmod installed.

For Vortex, this is already enforced in deployment, but we may need to account for it in load order and possible corner cases and manually overwritten data (although in those cases the files should be getting replaced which means we get the warning of external modification.)

### CODE REVIEW

**Comment by effs** [comment]
This thing's a mess with all the refactors. It's probably best to mainly look at:

- How installers.archive handles the REDmod transform (I still need to move the functions somewhere else but it's self-contained for now)
- How the ModInfo stuff is set up in installers.shared
- And how ModInfo is used in installers.ts (and not yet used by the other installers that variously rely on the path or the mod name... shouldn't be too big of a refactor, just need to hit every spot.)

**Review by E1337Kat — COMMENTED** [review]
There is a _whoooole_ lot going on here, but I like a lot of it. Def need to think about audio .archive mods, as those can not be simply autoconverted as the info.json actually requires more info than just a name and version. They need the customSounds array to be filled out with the correct info, and unless we can parse and infer such info from the .archive, I don't see us being able to handle them... if we can though.... I think it would be quite good. It would be kinda cool if we could determine what files are actually being used in the archives. so that we could do like LOOT stuff, you know? Anyways, a few small things throughout the files that bug me, but mostly good work!

File: src/installers.ts [hunk 1]
```diff
@@ -602,8 +611,8 @@ const installUsingPipelineOfInstallers: VortexWrappedInstallFunc = async (
   sourceDirPathForMod: string,
   stagingDirPathForMod: string,
   modName: string,
-  choices?: unknown,
-  unattended?: boolean,
+  modInfo: ModInfo,
+  features: Features,
```
[the remark this answers is not in the document]
Comment (effs) [reply src/installers.ts:615]: Yeah, I was thinking I'd clean that up too

File: test/unit/mods.example.archive.ts [hunk 2]
```diff
@@ -399,25 +402,26 @@ const ValidExtraArchivesWithTypeSucceeds = new Map<string, ExampleSucceedingMod>
 // REDmodding
 //
 
-const REDMOD_AUTOCONVERT_ARCHIVES: Features = {
+const FLAG_ENABLED_REDMOD_AUTOCONVERT_ARCHIVES: Features = {
   ...CurrentFeatureSet,
   REDmodAutoconvertArchives: Feature.Enabled,
 };
 
 // const REDMOD_AUTOCONVERT_MOD_NAME = `${FAKE_MOD_NAME} ${REDMOD_AUTOCONVERTED_NAME_TAG}`;
```
Comment (E1337Kat) [inline test/unit/mods.example.archive.ts:410]:
```suggestion
```
debug cruft removal

**Review by E1337Kat — COMMENTED** [review]
Once again, github makes looking at the diffs unnecessarily difficult, but this is good. I like how clean the wrapped test supported and installers are now. I like that whether autoconvert happens or not, a lot of the "nice to haves" here are added.

File: src/installers.ts [hunk 3]
```diff
@@ -19,36 +18,101 @@ import {
...
-import { testForCetMod, installCetMod } from "./installer.cet";
-import { testForIniMod, installIniMod } from "./installer.config.ini-reshade";
-import { testForJsonMod, installJsonMod } from "./installer.config.json";
-import { installConfigXmlMod, testForConfigXmlMod } from "./installer.config.xml";
-import { testForRed4ExtMod, installRed4ExtMod } from "./installer.red4ext";
-import { testForRedscriptMod, installRedscriptMod } from "./installer.redscript";
+import {
+  installCoreTweakXL,
+  testForCoreTweakXL,
+} from "./installer.core-tweak-xl";
```
[the remark this answers is not in the document]
Comment (effs) [reply src/installers.ts:41]: Yeah, could try to come up with some pattern for it (there's both an item and a line length toggle) but inasmuch as those are read at all, feels better if it's all the same? Easier to delete unused etc.

**Review by E1337Kat — APPROVED** [review]

**Comment by E1337Kat** [comment]
Need to go ahead and merge allllllllllllllllllllll of this now, because more work will just get even more difficult to get through.

**Pull request merged**